aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageDownloadAccessLevelTest.java

```diff
@@ -0,0 +1,232 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Instrumentation test to confirm that Storage Download behaves
+ * correctly with regards to the provided storage access level.
+ */
+public final class AWSS3StorageDownloadAccessLevelTest extends StorageInstrumentationTestBase {
+
+    private static final String USER_NAME_ONE = "test-user-1";
+    private static final String USER_NAME_TWO = "test-user-2";
+    private static Map<String, String> identityIds = new HashMap<>();
+
+    private static final String UPLOAD_NAME = "upload-test-" + System.currentTimeMillis();
+    private static final long UPLOAD_SIZE = 100L;
+    private static File uploadFile;
+
+    private final String destination = "download-test-" + System.currentTimeMillis();
+    private File downloadFile;
+
+    /**
+     * Upload the required resources in cloud prior to running the tests.
+     * @throws Exception if there is a problem while creating or uploading
+     *         the files.
+     */
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        // Randomly write a file to upload
+        uploadFile = createTempFile(UPLOAD_NAME, UPLOAD_SIZE);
+
+        // Upload to each access level
+        latchedUploadAndConfirm(uploadFile, StorageAccessLevel.PUBLIC, getIdentityId());
+
+        signInAs(USER_NAME_ONE);
+        identityIds.put(USER_NAME_ONE, getIdentityId());
+        latchedUploadAndConfirm(uploadFile, StorageAccessLevel.PROTECTED, getIdentityId());
+        latchedUploadAndConfirm(uploadFile, StorageAccessLevel.PRIVATE, getIdentityId());
+
+        signInAs(USER_NAME_TWO);
+        identityIds.put(USER_NAME_TWO, getIdentityId());
+        latchedUploadAndConfirm(uploadFile, StorageAccessLevel.PROTECTED, getIdentityId());
+        latchedUploadAndConfirm(uploadFile, StorageAccessLevel.PRIVATE, getIdentityId());
+    }
+
+    /**
+     * Clean up the uploaded resources after the test.
+     */
+    @AfterClass
+    public static void cleanUp() {
+        // Clean up each access level
+        cleanUpS3Object(getS3Key(StorageAccessLevel.PUBLIC, UPLOAD_NAME));
+
+        signInAs(USER_NAME_ONE);
+        cleanUpS3Object(getS3Key(StorageAccessLevel.PROTECTED, UPLOAD_NAME));
+        cleanUpS3Object(getS3Key(StorageAccessLevel.PRIVATE, UPLOAD_NAME));
+
+        signInAs(USER_NAME_TWO);
+        cleanUpS3Object(getS3Key(StorageAccessLevel.PROTECTED, UPLOAD_NAME));
+        cleanUpS3Object(getS3Key(StorageAccessLevel.PRIVATE, UPLOAD_NAME));
+
+        identityIds.clear();
+    }
+
+    /**
+     * Signs out by default and sets up the file to download object to.
+     * @throws Exception if an error is encountered while creating file
+     */
+    @Before
+    public void setUp() throws Exception {
+        signOut();
+        downloadFile = createTempFile(destination);
```

Comment:
well upload file is instantiated in `@BeforeClass` so that it can only be uploaded one time to reduce heavy workload. Technically download file can be created once also, but this kind of just serves to show that each there is a file being downloaded in each test case (as opposed to one uploaded file in the entire test suite). This is no big deal and I did it without any major reason though, so I can change it if you do not like it this way.

Reply:
Either way - this one wasn't isn't a big deal.